tests/ts_updown_util.erl

```diff
@@ -0,0 +1,361 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2016 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%%
+%% @doc Facilities to use in CT-enabled upgrade/downgrade tests.
+
+-module(ts_updown_util).
+
+-export([
+         make_config/1,
+         maybe_shutdown_client_node/1,
+         run_scenarios/2,
+         run_scenario/2
+        ]).
+
+-include_lib("eunit/include/eunit.hrl").
+-include("ts_updown_util.hrl").
+
+-type versioned_cluster() :: [{node(), version()}].
+%% preparations
+
+make_config(Config) ->
```

Comment:
This seems to be misnamed - the config is a side-effect of building the cluster - maybe rename this function `build_cluster_and_and_return_config`